server/jwt/views.py

```diff
@@ -0,0 +1,21 @@
+from django.core.exceptions import PermissionDenied
+from django.http import JsonResponse
+from django.views.decorators.csrf import ensure_csrf_cookie
+from rest_framework_simplejwt.tokens import RefreshToken
+
+
+@ensure_csrf_cookie
+def obtain_token(request):
+    """
+    Obtains a token for an already-authorized user
+
+    rest_frame_simplejwt by default requires a username/password post blob for
+    its token-obtaining endpoint (which we usually aren't in a position to provide),
+    so we provide this simple method which does the equivalent, using Django's
+    existing session-based authentication to identify and validate the user.
+    """
+    if not request.user.is_authenticated:
+        raise PermissionDenied
+
+    refresh = RefreshToken.for_user(request.user)
+    return JsonResponse(data={"refresh": str(refresh), "access": str(refresh.access_token)})
```

Comment:
I understand why we'd not want to use the built-in view, but I think it would be better to implement an own `GenericAPIView` following the general API that `rest_framework_simplejwt` follows. The advantage being that we could use DRF's session auth backend just locally for that view.

Reply:
This sounds like a good idea, I will give it a go

Reply:
Implemented, we'll continue discussion of the server pieces in #1906